ui/appui-react/src/appui-react/frontstage/FrontstageConfig.ts

```diff
@@ -0,0 +1,73 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Frontstage
+ */
+
+import { CommonProps } from "@itwin/core-react";
+import { ContentGroup, ContentGroupProvider } from "../content/ContentGroup";
+import { ToolItemDef } from "../shared/ToolItemDef";
+import { PanelConfig } from "../stagepanels/PanelConfig";
+import { WidgetConfig } from "../widgets/WidgetConfig";
+
+/** Configuration from which a frontstage is created.
+ * @public
```

Comment:
Can't ATM, since `FrontstageProvider.frontstage` is marked as **public**..
Returning that union type from the `FrontstageProvider.frontstage` is also breaking, I think we should also add an optional getter that would return a new config instead. Any suggestions on the naming?

Reply:
Well, technically it's a public class, so we have to assume that someone might use it outside of our code base.
Until we decide on the naming I have added a separate method to get the config: `public frontstageConfig?(): FrontstageConfig;`.